### PR TITLE
added DAB support

### PR DESCRIPTION
**Pull request by initializesahib**
[dab](https://github.com/InitializeSahib/dab) is the **hot** :tm: new build tool everyone's raving about  
this PR adds support for DAB via a Dabfile (to accentuate the low-level meme)

### CODE REVIEW

File: Dabfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+build() {
```
Comment (initializesahib) [inline Dabfile:1]: Everything from this point to the next comment was copied verbatim from meke.sh, excluding the extra indentation

File: Dabfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	nasm bootloader/bootloader.asm -f bin -o me.img
+	dd bs=32768 count=256 if=/dev/zero >> me.img
+	truncate --size=-4096 me.img
+	./echfs-utils me.img format 512
+	./cc/cc32 kernel/drivers/graphics.c
+	nasm userland/generic_program.asm -f bin -o generic_program.bin
+	./echfs-utils me.img import generic_program.bin generic_program.bin
+	nasm kernel/stub.asm -f bin -o me.bin
+	./echfs-utils me.img import me.bin me.bin
+	return 0
```
Comment (initializesahib) [inline Dabfile:20]: This is the second new line.

File: Dabfile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+}
+default() {
```
Comment (initializesahib) [inline Dabfile:22]: This task is required so `dab` can run without `-t build`.

**Comment by dm4uz3** [comment]
nigger

File: Dabfile (hunk 1 above)
Comment (dm4uz3) [reply Dabfile:1]: ni

File: Dabfile (hunk 2 above)
Comment (dm4uz3) [reply Dabfile:20]: g

File: Dabfile (hunk 3 above)
Comment (dm4uz3) [reply Dabfile:22]: ger

**Pull request merged**